File: app/src/main/java/com/mancode/financetracker/ui/accounts/AccountFragment.java
package com.mancode.financetracker.ui.accounts;

import android.content.Context;
import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import com.google.android.material.floatingactionbutton.FloatingActionButton;
import com.mancode.financetracker.R;
import com.mancode.financetracker.database.viewmodel.AccountViewModel;
import com.mancode.financetracker.ui.UIUtils;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.fragment.app.Fragment;
import androidx.lifecycle.ViewModelProviders;
import androidx.recyclerview.widget.LinearLayoutManager;
import androidx.recyclerview.widget.RecyclerView;

/**
 * Created by Manveru on 03.09.2017.
 */

public class AccountFragment extends Fragment {

    private AccountRecyclerViewAdapter mAdapter;

    public AccountFragment() { }

    public static AccountFragment newInstance() {
        return new AccountFragment();
    }

    @Override
    public void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        AccountViewModel accountViewModel =
                ViewModelProviders.of(this).get(AccountViewModel.class);
        accountViewModel.getAllAccounts().observe(this,
                accounts -> mAdapter.setAccounts(accounts));
    }

    @Override
    public View onCreateView(@NonNull LayoutInflater inflater, ViewGroup container,
                             Bundle savedInstanceState) {
        View view = inflater.inflate(R.layout.fragment_account_list, container, false);
        View recyclerView = view.findViewById(R.id.list);

        // Set the adapter
        if (recyclerView instanceof RecyclerView) {
            Context context = view.getContext();
            RecyclerView rView = (RecyclerView) recyclerView;
            rView.setLayoutManager(new LinearLayoutManager(context));
            mAdapter = new AccountRecyclerViewAdapter();
            rView.setAdapter(mAdapter);
        }

        FloatingActionButton fab = view.findViewById(R.id.fab);
        fab.setOnClickListener(view1 -> UIUtils.showFullScreenDialog(
                getFragmentManager(), AddAccountFragment.newInstance()));
        return view;
    }
}

File: app/src/main/java/com/mancode/financetracker/ui/accounts/AccountFragment.kt
package com.mancode.financetracker.ui.accounts

import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.fragment.app.Fragment
import androidx.fragment.app.viewModels
import androidx.navigation.NavController
import androidx.navigation.fragment.findNavController
import androidx.recyclerview.widget.LinearLayoutManager
import com.mancode.financetracker.R
import com.mancode.financetracker.databinding.FragmentAccountListBinding
import com.mancode.financetracker.viewmodel.AccountViewModel

/**
 * Created by Manveru on 03.09.2017.
 */

class AccountFragment : Fragment(R.layout.fragment_account_list), AccountRecyclerViewAdapter.ModifyRequestListener {

    private var _binding: FragmentAccountListBinding? = null
    private val binding get() = _binding!!

    lateinit var navController: NavController
    private val adapter: AccountRecyclerViewAdapter by lazy { AccountRecyclerViewAdapter(this) }
    private val viewModel: AccountViewModel by viewModels()

    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?,
                              savedInstanceState: Bundle?): View {
        _binding = FragmentAccountListBinding.inflate(inflater, container, false)
        return binding.root
    }

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)
        viewModel.allAccounts.observe(viewLifecycleOwner, { accounts ->
                    adapter.setAccounts(accounts)
                    if (accounts.isNotEmpty()) {
                        binding.accountsList.visibility = View.VISIBLE
                        binding.emptyListInfo.visibility = View.GONE
                    } else {
                        binding.accountsList.visibility = View.GONE
                        binding.emptyListInfo.visibility = View.VISIBLE
                    }
                })
        navController = findNavController()
        binding.accountsList.layoutManager = LinearLayoutManager(context)
        binding.accountsList.adapter = adapter
        binding.fab.setOnClickListener {
            navController.navigate(R.id.action_accountFragment_to_addEditAccountFragment)
        }
    }

    override fun onDestroyView() {
        _binding = null
        super.onDestroyView()
    }

    override fun onEditRequested(accountId: Int) {
        val action = AccountFragmentDirections.actionAccountFragmentToAddEditAccountFragment(accountId)
        navController.navigate(action)
    }
}
